pandas/core/construction.py

```diff
@@ -212,29 +221,20 @@ def array(
     [a, b, a]
     Categories (3, object): [a < b < c]
 
-    Because omitting the `dtype` passes the data through to NumPy,
-    a mixture of valid integers and NA will return a floating-point
-    NumPy array.
+    If pandas does not infer a dedicated extension type a
+    :class:`arrays.PandasArray` is returned.
```

Comment:
Should we mention that this can still change in the future? (eg that more types start to get inferred, so basically that you should not rely on the fact of `pd.array` returning a PandasArray when no dtype is specified)